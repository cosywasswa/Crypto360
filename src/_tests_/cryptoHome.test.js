import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import CryptoHome from '../components/cryptoHome';

const mockStore = configureStore([]);

describe('cryptoHome component', () => {
  const mockcryptos = [
    {
      key: 'bitcoin',
      id: 'bitcoin',
      rank: '1',
      symbol: 'BTC',
      name: 'Bitcoin',
    },
    {
      key: 'ethereum',
      id: 'ethereum',
      rank: '2',
      symbol: 'ETH',
      name: 'Ethereum',
    },
  ];

  const Store = mockStore({
    cryptoList: {
      cryptoArr: mockcryptos,
      loading: false,
      isError: false,
      searchQuery: '',
    },
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(

        <Provider store={Store}>
          <BrowserRouter>
            <CryptoHome />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
