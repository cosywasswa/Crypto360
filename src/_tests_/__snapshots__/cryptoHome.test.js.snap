// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cryptoHome component should match the snapshot 1`] = `
<div
  className="crypto-container"
>
  <div
    className="logo-div"
  >
    <img
      alt="crypto "
      className="logo"
      src="crypto.png"
    />
  </div>
  <div
    className="stat-and-form"
  >
    <h1
      className="stats"
    >
      Crypto Stats
    </h1>
    <form
      className="search-form"
    >
      <input
        className="input-field"
        onChange={[Function]}
        placeholder="Search for Currency "
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="main-item"
  >
    <div
      className="container"
    >
      <a
        className="crypto"
        href="/Details/bitcoin"
        onClick={[Function]}
      >
        <div
          className="crypto-item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle-arrow-right arrow"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h2>
            BTC
          </h2>
          <h4>
            Bitcoin
          </h4>
          <p>
            Rank#
            1
          </p>
        </div>
      </a>
      <a
        className="crypto"
        href="/Details/ethereum"
        onClick={[Function]}
      >
        <div
          className="crypto-item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle-arrow-right arrow"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h2>
            ETH
          </h2>
          <h4>
            Ethereum
          </h4>
          <p>
            Rank#
            2
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
